### PR TITLE
Output consistency tests: various improvements

### PR DESCRIPTION
**Pull request by nrainer-materialize**
Some refactorings and improvements (that awaited the merge of the timestamp functionality), most from last week.

### Tips for reviewer
* commit-wise
* ✅ https://github.com/MaterializeInc/materialize/pull/19671 needs to be merged first

### CODE REVIEW

File: misc/python/materialize/output_consistency/operation/operation.py [hunk 1]
```diff
@@ -42,7 +42,12 @@ def __init__(
         args_validators: Optional[Set[OperationArgsValidator]] = None,
         is_aggregation: bool = False,
         relevance: OperationRelevance = OperationRelevance.DEFAULT,
+        is_disabled: bool = False,
```
Comment (def-) [inline misc/python/materialize/output_consistency/operation/operation.py:45]: I prefer positive messages, so `is_enabled: bool = True`.

File: misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py [hunk 2]
```diff
@@ -32,6 +32,20 @@
 )
 
 
+class IgnoreVerdict:
+    def __init__(self, ignore: bool, reason: Optional[str]):
+        self.ignore = ignore
+        self.reason = reason
+
+
+def yes_ignore(reason: str) -> IgnoreVerdict:
```
Comment (def-) [inline misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py:41]: Rust's sum types would be perfect for this. You could use `@dataclass`

**Review by def- — APPROVED** [review]
Just some minor nitpicks, looks good

**Comment by def-** [comment]
Would be cool to get nightly green btw: `function age(timestamp, time) does not exist', 'H': 'No function matches the given name and argument types. You might need to add explicit type casts.`

**Comment by nrainer-materialize** [comment]
> Would be cool to get nightly green btw: `function age(timestamp, time) does not exist', 'H': 'No function matches the given name and argument types. You might need to add explicit type casts.`

Yes, thanks for reminding me.
I noticed this error earlier this week and had thought it might have resolved itself. The issue is simply that the docs on our website are ahead; the `age` function was recently added to the docs but is apparently not available yet. I will disable it.

Furthermore, I need to look into why this error breaks the whole test run... it shouldn't

**Pull request merged**